packages/mobile/src/transactions/TransactionFeed.tsx

```diff
@@ -103,24 +71,58 @@ export class TransactionFeed extends React.PureComponent<Props> {
     }
   }
 
-  keyExtractor = (item: TransactionFeedFragment) => {
+  const renderSectionHeader = (info: { section: SectionListData<FeedItem> }) => (
+    <SectionHeadNew text={info.section.title} />
+  )
+
+  const keyExtractor = (item: TransactionFeedFragment) => {
     return item.hash + item.timestamp.toString()
   }
 
-  render() {
-    const { kind, loading, error, data } = this.props
-
-    if (error) {
-      Logger.error(TAG, 'Failure while loading transaction feed', error)
-      return <NoActivity kind={kind} loading={loading} error={error} />
+  const sections = useMemo(() => {
+    // Only compute sections for home screen.
+    if (!data || data.length === 0 || kind !== FeedType.HOME) {
+      return []
     }
+    return groupFeedItemsInSections(data)
+  }, [kind, data])
 
-    if (data && data.length > 0) {
-      return <FlatList data={data} keyExtractor={this.keyExtractor} renderItem={this.renderItem} />
-    } else {
-      return <NoActivity kind={kind} loading={loading} error={error} />
-    }
+  if (error) {
+    Logger.error(TAG, 'Failure while loading transaction feed', error)
+    return <NoActivity kind={kind} loading={loading} error={error} />
+  }
+
+  if (!data || data.length === 0) {
+    return <NoActivity kind={kind} loading={loading} error={error} />
+  }
+
+  if (kind === FeedType.HOME) {
+    return (
+      <SectionList
+        renderItem={renderItem}
+        renderSectionHeader={renderSectionHeader}
+        sections={sections}
+        keyExtractor={keyExtractor}
+        keyboardShouldPersistTaps="always"
+      />
+    )
+  } else {
+    return <FlatList data={data} keyExtractor={keyExtractor} renderItem={renderItem} />
   }
 }
 
-export default connect<StateProps, {}, {}, RootState>(mapStateToProps)(TransactionFeed)
+export const TransactionFeedFragments = {
+  transaction: gql`
+    fragment TransactionFeed on TokenTransaction {
+      ...ExchangeItem
+      ...TransferItem
+    }
+
+    ${ExchangeFeedItem.fragments.exchange}
+    ${TransferFeedItem.fragments.transfer}
+  `,
+}
+
+// TODO: We should do a better comparison of the |data| prop to decide when to re-render.
+// Right now |data| always changes (returns a different ref) which causes many extra renders.
+export default memo(TransactionFeed)
```

Comment:
Maybe we shouldn't use `memo` for now then?

Reply:
I guess that makes sense, I used it because we were using `PureComponent` and wanted to maintain an equivalence to that, but I'll remove it :)